packages/smart-vaults/src/library/MultiSignerSignatureLib.sol

```diff
@@ -135,37 +134,45 @@ library MultiSignerSignatureLib {
         }
     }
 
-    function getSignerAtIndex(bytes memory signers_, uint8 index_) internal pure returns (bytes memory, uint8) {
-        uint256 start = uint256(index_) * SIGNER_SIZE;
-
-        uint8 signerType;
-        assembly {
-            signerType := byte(0, mload(add(signers_, add(32, start))))
-        }
-
-        uint256 returnLength;
-        if (signerType == EMPTY_SIGNER_TYPE || signerType == REMOVED_SIGNER_TYPE) {
-            return (new bytes(0), signerType);
-        } else if (signerType == EOA_SIGNER_TYPE) {
-            returnLength = MultiSignerLib.EOA_SIGNER_SIZE;
-        } else if (signerType == PASSKEY_SIGNER_TYPE) {
-            returnLength = MultiSignerLib.PASSKEY_SIGNER_SIZE;
-        }
-
-        bytes memory signer = new bytes(returnLength);
+    function getSignerAtIndex(bytes memory signerUpdates_, uint8 index_) internal pure returns (bytes memory) {
```

Comment:
i think this fn can be cleaned up a bit but dont want to hold up submitting this review for it